### ISSUE TITLE
Compilation Error

### ISSUE THREAD

**Issue by SanderPaulus** [Hieromon/AutoConnect#125]
Good evening,

I'am trying to compile a sketch which gives me the following error nd I am not sure what to do to solve this issue. Any guidance would be wildly appreciated!

Multiple libraries were found for "WiFi.h"
 Used: C:\Users\sande\Documents\ArduinoData\packages\esp32\hardware\esp32\1.0.2\libraries\WiFi
 Not used: C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.21.0_x86__mdqgnx93n4wtt\libraries\WiFi
Multiple libraries were found for "WebServer.h"
 Used: C:\Users\sande\Documents\ArduinoData\packages\esp32\hardware\esp32\1.0.2\libraries\WebServer
 Not used: C:\Users\sande\Documents\Arduino\libraries\SDL_ESP32_BC24_GETIP-master
exit status 1
Error compiling for board DOIT ESP32 DEVKIT V1.

**Comment by Hieromon** [comment]
```
Multiple libraries were found for "WiFi.h"
Used: C:\Users\sande\Documents\ArduinoData\packages\esp32\hardware\esp32\1.0.2\libraries\WiFi
Not used: C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.21.0_x86__mdqgnx93n4wtt\libraries\WiFi
```
It is a known issue. Please refer to https://forum.arduino.cc/index.php?topic=488732.0 and https://github.com/espressif/arduino-esp32/issues/20

**Comment by SanderPaulus** [comment]
Good evening Hieromon,

> Please post the type of ESP32 board you are using 
I am using the DOIT ESP32 DEVKIT V1

> and the log of the upload phase from esptool.
Not sure what you mean by that, but I will give you the output of the debugger window
`Arduino: 1.8.9 (Windows 10), Board: "DOIT ESP32 DEVKIT V1, 80MHz, 921600, None"

Sketch uses 631346 bytes (48%) of program storage space. Maximum is 1310720 bytes.
Global variables use 37960 bytes (11%) of dynamic memory, leaving 289720 bytes for local variables. Maximum is 327680 bytes.
esptool.py v2.6
Serial port COM8
Connecting........_____....._____....._____....._____....._____....._____.....____An error occurred while uploading the sketch
_

A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.
`

I don't think it's a USB driver issue as I was saying it worked before which also makes it not a power related issue.
To make things even more weird(by lack of a better word). I just removed the ESP32 from the board manager, and simply reinstalled it. Now it will compile and upload the WiFiScan sketch, but won't compile anything from the autoconnect library.

When I was trying to upload the Simple sketch before removing the board and installing it, there was a blinking blue led when arduino ide was trying to connect to the board. Just now while uploading the WiFiScan sketch there was no blinking at all....

Kind regards

**Comment by Hieromon** [comment]
> I am using the DOIT ESP32 DEVKIT V1

Schematic is here. https://www.dropbox.com/s/jefwxxtufgwg0ex/esp32_Schematic%20Prints.pdf?dl=0

The esptool error log suggests that ESP32 is not in flash mode. Also, it circuit requires you to hold down the "Boot" button when flashing ESP32. Did you?
Or, Did you connect 10uF capacitor between EN and GND to go into flashing/uploading automatically?
In any case, the issue is unrelated to AutoConnect.

**Comment by SanderPaulus** [comment]
Hi Hieromon,

I think there is a misunderstanding somewhere, perhaps I did not explain it well enough.

So at this moment I cannot compile any Autoconnect sketches with the same error I started this issue. 
However I can compile and upload the WiFiScan sketch from the WiFi library, without any issues without holding down the boot button or connecting a capacitor. The sketch works then as expected! Which suggests to me that the hardware and USB drivers are in working order. 
The order of installing libraries and boards for the above is as follow: Pagebuilder, ArduinoJson, Autoconnect then the ESP32 board.

If I would remove the libraries and boards, and install them in the following order: ESP32 board, pagebuilder,ArduinoJson then Autoconnect I can no longer compile the WifiScan sketch but I can compile all Autoconnect sketches but then I can no longer upload the sketches because the board does not go into flash mode. Basically the other way around.

So the order of installing libraries and boards seems to make all the difference.

Hopefully this clarify's the situation.

So for our amusement I can try the capacitor "hack" to see if this is the solution for my problems.

Anyway, thank you very much for trying to help me!

**Comment by Hieromon** [comment]
> So the order of installing libraries and boards seems to make all the difference.

I don't think so. There is no relationship between the library installation order and the boot mode of ESP32 module.
As a specification of esptool, it supports two way for entering the boot loader modes.
https://github.com/espressif/esptool/wiki/ESP32-Boot-Mode-Selection#select-bootloader-mode
1. GPIO0 input as LOW
2. Using the RTS and DTR modem status lines to toggle GPIO0 and EN

GPIO0 is pulling up to VDD3V3 as shown in the schematic I posted the above. And EN connects to GPIO0 via RTS and DTR. However, the time to maintain the signal is too shorter to enter boot mode stably. (It may be okay if you're lucky, but it's not always)
So, you need to make enough time for esptool to enter boot mode with a 10uf capacitor.
Here is the reference for you. https://randomnerdtutorials.com/solved-failed-to-connect-to-esp32-timed-out-waiting-for-packet-header/

> BTW, this is my latest error I received...
> error: 'class Preferences' has no member named 'getBytesLength'

Please use a arduino-esp32 1.0.2.
The getBytesLength is not implemented  arduino-esp32 core 1.0.1 or earlier. For this, I will release a patch at an early stage so that it can be executed with the past esp32 core. For the immediate workaround, use arduino-esp32 1.0.2.
